controllers/nstemplateset/nstemplateset_controller.go

```diff
@@ -136,12 +136,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	} else if createdOrUpdated {
 		return reconcile.Result{}, nil // wait for cluster resources to be created
 	}
+	if err := r.status.updateStatusClusterResourcesRevisions(ctx, nsTmplSet); err != nil {
```

Comment:
Do we need to update the revisions 3 times during the reconcile? I get that we do it after we apply the 3 different kinds of resources from the template set but apart from maybe easier state management in case of errors, is there any benefit to doing it this way instead of just once at (each) end of the reconcile method beyond this point?

Reply:
As Matous said this seemed to be the most straightforward way to store the information. Also, IMO it is more accurate to update the status as soon as we are done with one type/kind of template, since things might go wrong later in the code and we don't have to start again from scratch , especially when the update of the namespace will rely on this comparison logic that I'm introducing.